test/smartwallet/smartWallet.test.ts

```diff
@@ -175,18 +179,242 @@ describe('SmartWallet contract', function () {
+          .called;
+      });
+
+      it('Should not pay on sponsored deployment', async function () {
+        const deployRequest = createDeployRequest({
+          relayHub: relayHub.address,
+          from: owner.address,
+          nonce: '0',
+          tokenGas: '0',
+          tokenAmount: '0',
+          tokenContract: fakeToken.address,
+          gas: '4000',
+          to: recipient.address,
+          data: recipientFunction,
+        });
+
+        const typedDeployData = new TypedDeployRequestData(
+          HARDHAT_CHAIN_ID,
+          smartWalletFactory.address,
+          deployRequest
+        );
+
+        const suffixData = getSuffixData(typedDeployData);
+
+        const privateKey = Buffer.from(
+          owner.privateKey.substring(2, 66),
+          'hex'
+        );
+        const signature = getLocalEip712DeploySignature(
+          typedDeployData,
+          privateKey
+        );
+
+        const workerBalanceBefore = await fakeToken.balanceOf(worker.address);
+
+        await smartWalletFactory
+          .connect(relayHub)
+          .relayedUserSmartWalletCreation(
+            deployRequest.request,
+            suffixData,
+            worker.address,
+            signature
+          );
+
+        const workerBalanceAfter = await fakeToken.balanceOf(worker.address);
+
+        expect(workerBalanceBefore).to.be.equal(workerBalanceAfter);
+        expect(fakeToken.transfer).not.to.be.called;
+        expect(recipient.isInitialized, 'Recipient method was not called').to.be
+          .called;
+      });
+
+      it('Should fail if contract execution fail', async function () {
+        const deployRequest = createDeployRequest({
+          relayHub: relayHub.address,
+          from: owner.address,
+          nonce: '0',
+          tokenGas: '0',
+          tokenAmount: '0',
+          tokenContract: fakeToken.address,
+          gas: '4000',
+          to: recipient.address,
+          data: recipientFunction,
+        });
+
+        const typedDeployData = new TypedDeployRequestData(
+          HARDHAT_CHAIN_ID,
+          smartWalletFactory.address,
+          deployRequest
+        );
+
+        const suffixData = getSuffixData(typedDeployData);
+
+        const privateKey = Buffer.from(
+          owner.privateKey.substring(2, 66),
+          'hex'
+        );
+        const signature = getLocalEip712DeploySignature(
+          typedDeployData,
+          privateKey
+        );
+
+        recipient.isInitialized.reverts('Failing in purpose');
+
+        await expect(
+          smartWalletFactory
+            .connect(relayHub)
+            .relayedUserSmartWalletCreation(
+              deployRequest.request,
+              suffixData,
+              worker.address,
+              signature
+            )
+        ).to.be.rejectedWith('Unable to execute');
+
+        expect(recipient.isInitialized, 'Recipient method was not called').to.be
```

Comment:
I'd also check that the revert message corresponds to `Failing in purpose`

Reply:
updated